cmd/pipeline/main.go

```diff
@@ -467,3 +468,23 @@ func main() {
 		router.Run(bindAddr)
 	}
 }
+
+func startInternalAPIRouter(skipPaths []string, db *gorm.DB, basePath string, clusterAPI *api.ClusterAPI) {
+	//Initialise Gin router for Internal API
+	internalRouter := gin.New()
+	internalRouter.Use(correlationid.Middleware())
+	internalRouter.Use(ginlog.Middleware(log, skipPaths...))
+	internalRouter.Use(gin.Recovery())
+	if viper.GetBool("audit.enabled") {
+		log.Infoln("Audit enabled, installing Gin audit middleware to internal router")
+		internalRouter.Use(audit.LogWriter(skipPaths, viper.GetStringSlice("audit.headers"), db, log))
+	}
+	internalGroup := internalRouter.Group(path.Join(basePath, "api", "v1/", "orgs"))
+	internalGroup.Use(auth.InternalUserHandler)
+	internalGroup.Use(api.OrganizationMiddleware)
+	internalGroup.GET("/:orgid/clusters/:id/nodepools", api.GetNodePools)
+	internalGroup.PUT("/:orgid/clusters/:id/nodepools", clusterAPI.UpdateNodePools)
+	internalBindAddr := viper.GetString("pipeline.internalBindAddr")
+	logger.Infof("Pipeline internal API listening on http://%s", internalBindAddr)
+	go internalRouter.Run(internalBindAddr)
```

Comment:
I would prefer the router being returned and the go routine being started somewhere in main.